packages/utils/src/normalize.ts

```diff
@@ -100,8 +100,17 @@ function visit(
     return value as ObjOrArray<unknown>;
   }
 
+  // Do not normalize objects that we know have already been normalized. As a general rule, the
+  // "__sentry_skip_normalization__" property should only be used sparingly and only should only be set on objects that
+  // have already been normalized.
+  let overriddenDepth = depth;
+
+  if ((value as ObjOrArray<unknown>)['__sentry_override_normalization_depth__']) {
```

Comment:
l: Maybe `if typeof value['...'] === 'number'`? Then we can remove the type cast below (I think?) and at least avoid this being incorrect. maybe overkill, though...